lib/Backend/amd64/LinearScanMD.cpp

```diff
@@ -450,3 +450,93 @@ RegNum LinearScanMD::GetRegisterFromSaveIndex(uint offset)
 {
     return (RegNum)(offset >= RegXMM0 ? (offset - RegXMM0) / (sizeof(SIMDValue) / sizeof(Js::Var)) + RegXMM0 : offset);
 }
+
+RegNum LinearScanMD::GetParamReg(IR::SymOpnd *symOpnd, Func *func)
+{
+	RegNum reg = RegNOREG;
+	StackSym *paramSym = symOpnd->m_sym->AsStackSym();
+
+	if (func->GetJnFunction()->GetIsAsmjsMode() && !func->IsLoopBody())
+	{
+		// Asm.js function only have 1 implicit param as they have no CallInfo, and they have float/SIMD params.
+		// Asm.js loop bodies however are called like normal JS functions.
+		if (IRType_IsFloat(symOpnd->GetType()) || IRType_IsSimd(symOpnd->GetType()))
+		{
+			switch (paramSym->GetParamSlotNum())
+			{
+			case 10000:
+				reg = RegXMM0;
+				break;
+			case 1:
+				reg = RegXMM1;
+				break;
+			case 2:
+				reg = RegXMM2;
+				break;
+			case 3:
+				reg = RegXMM3;
+				break;
+			}
+		}
+		else
+		{
+			if (paramSym->IsImplicitParamSym())
+			{
+				switch (paramSym->GetParamSlotNum())
+				{
+				case 1:
+					reg = RegRCX;
+					break;
+				default:
+					Assert(UNREACHED);
+				}
+			}
+			else
+			{
+				switch (paramSym->GetParamSlotNum())
+				{
+				case 1:
+					reg = RegRDX;
+					break;
+				case 2:
+					reg = RegR8;
+					break;
+				case 3:
+					reg = RegR9;
+					break;
+				}
+			}
+		}
+	}
+	else // Non-Asm.js
+	{
+		Assert(symOpnd->GetType() == TyVar || IRType_IsNativeInt(symOpnd->GetType()));
+
+		if (paramSym->IsImplicitParamSym())
+		{
+			switch (paramSym->GetParamSlotNum())
```

Comment:
They just get loaded from the stack for now.  We can fix this when optimizations are enabled for generators.

---
In reply to: [75064004](https://github.com/Microsoft/ChakraCore/pull/1373#discussion_r75064004) [](ancestors = 75064004)